### ISSUE TITLE
The editor should have a url that I can share that shows the resource.

### ISSUE THREAD

**Issue by jcoyne** [LD4P/sinopia_editor#2990]
So that I can share with someone else the page I'm editing.  

Currently I have to know the URI in the API and then search for that resource and then click on it. 

This may also be useful to link from Folio.

**Comment by michelleif** [comment]
yes that would be nice; we currently have URIs for sharing a **template**, but it's not working so is slated to be dumped see #2770

**Comment by michelleif** [comment]
we do have this, at the top of a resource screen.
![Screen Shot 2021-09-10 at 2.05.54 PM.png](https://images.zenhubusercontent.com/5d825bf74def500001d12e70/7dc13bf1-144f-4791-aedd-d6501ccc4ca4)

**Comment by justinlittman** [comment]
This will be useful in Folio so that a link can be provided that will open a resource in Sinopia.

**Comment by michelleif** [comment]
oh, i see, ignore my comments! (i will delete)

**Comment by justinlittman** [comment]
Currently, the Editor supports:
* For the URL `/editor/<template id>`
  * When a URL matching the pattern is entered in the browser, the editor is opened to a blank resource editing that template.
  * When a user opens an a new or existing resource in the editor, the URL is changed to match the pattern for the resource's template.
  * As the user switches between resources in the editor, the URL is changed to match the pattern for the current resource's template.

Please change this so that:
* For the URL `/editor/<template id>`
  * When a URL matching the pattern is entered in the browser:
    * If the template exists and the user has create permissions, the editor is opened to a blank resource editing that template.
    * If the template exists, but the user does not have create permissions, the URL is changed to `/dashboard` and the user receives an error message.
    *  If the template does not exist, the URL is changed to `/templates` and the user receives an error message.
  * When a user opens an a new or copied resource in the editor (i.e., unsaved), the URL is changed to match the pattern for the resource's template.
* For the URL `/editor/resource/<resource id>`
  * When a URL matching the pattern is entered in the browser:
    * If the resource exists and the user has edit permissions, the editor is opened to editing that resource.
    * If the resource exists and the user does not have edit permissions, the URL is changed to `/dashboard` and a preview modal is opened containing the resource.
    * If the resource does not exist, the URL is changed to `/dashboard` and the user receives an error message.
  * When a user opens an existing resource in the editor, the URL is changed to match the pattern for the resource.
  * When a user saves a resource (i.e., it gets a resource id), the URL is changed to match the pattern for the resource (i.e., it is changed from the template pattern).
  * As the user switches between resources in the editor, the URL is changed to match the pattern for the current resource's template (when new/copied/unsaved) or the current resource (when existing/saved).


Please also address #2770
